op-service/rpc/server.go

```diff
@@ -165,36 +173,41 @@ func (b *Server) AddAPI(api rpc.API) {
 }
 
 func (b *Server) Start() error {
-	srv := rpc.NewServer()
-
+	// Register all APIs to the RPC server.
 	for _, api := range b.apis {
-		if err := srv.RegisterName(api.Namespace, api.Service); err != nil {
+		if err := b.rpcServer.RegisterName(api.Namespace, api.Service); err != nil {
 			return fmt.Errorf("failed to register API %s: %w", api.Namespace, err)
 		}
 		b.log.Info("registered API", "namespace", api.Namespace)
 	}
 
-	// rpc middleware
-	var nodeHdlr http.Handler = srv
-	for _, middleware := range b.middlewares {
-		nodeHdlr = middleware(nodeHdlr)
-	}
-	nodeHdlr = node.NewHTTPHandlerStack(nodeHdlr, b.corsHosts, b.vHosts, b.jwtSecret)
+	// http handler stack.
+	var handler http.Handler
+
+	// default to 404 not-found
+	handler = http.HandlerFunc(http.NotFound)
+
+	// Health endpoint is lowest priority.
+	handler = b.newHealthMiddleware(handler)
+
+	// serve RPC on configured RPC path (but not on arbitrary paths)
+	handler = b.newHttpRPCMiddleware(handler)
```

Comment:
took me a moment based on the naming, but I see this follows an extend-and-return pattern, similar to `append`. :+1: